### PR TITLE
Add cache invalidation

### PR DESCRIPTION
**Pull request by amoghar29**
### Summary

This PR implements cache invalidation.

**Why do we need to invalidate CloudFront cache?**

- Whenever an image or file is uploaded, it gets cached and stored on CloudFront edge servers for faster loading and an improved user experience. However, if the same file is reuploaded or updated, the edge servers will continue serving the old, cached version to users because they are unaware of the changes.

- To ensure that users always receive the latest version of the file, we need to invalidate the existing cache. This forces the edge servers to fetch the updated content from the origin server, ensuring the latest version is served.

### Approach

In this PR, I have added functionality to invalidate the CloudFront cache.
- Created the cloudFrontInvalidate(paths) function to handle cache invalidation.

- This function takes an array of paths as input and sends an invalidation request to CloudFront using the AWS SDK’s CreateInvalidationCommand.

**How to invalidate cache?**

- import the cloudFrontInvalidate function from cloudFront.js
- pass an array containing the path of the file to be invalidated.
**Note -**  path should be in the format of :   /extension/filename.extension

**Additional Setup Steps in AWS**

1. Find the Distribution ID

Navigate to the CloudFront service in the AWS Management Console.
Locate your distribution under the Distributions section.
Copy the Distribution ID (you'll need this to set the DISTRIBUTION_ID environment variable).
Find the ARN

2. In the Details section of your selected distribution, you will find the ARN (Amazon Resource Name).

Copy the ARN (you'll need this to create the IAM policy).
Create an IAM Policy

3. Navigate to the IAM service in AWS Management Console.
Under Policies, click Create policy and select the JSON tab.
Paste the following into the policy editor, replacing replace with your ARN with the ARN you copied earlier:
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": "cloudfront:CreateInvalidation",
      "Resource": "replace with your ARN"
    }
  ]
}
```

### Screenshots or Recordings

https://github.com/user-attachments/assets/d6ccc87e-ae90-4dd2-bea4-7311b0379f29

## Checklist

- [ ] I have added/updated tests that cover the changes.
- [ ] I have updated the documentation to reflect the changes.

### CODE REVIEW

File: server/utils/cloudFront.js [hunk 1]
```diff
@@ -21,4 +25,39 @@ function cloudFrontSignedURL(path) {
...
+  region: process.env.BUCKET_REGION,
+  credentials: {
+    accessKeyId: process.env.ACCESS_KEY,
+    secretAccessKey: process.env.SECRET_ACCESS_KEY,
+  },
+};
+const client = new CloudFrontClient(config); // Initialize CloudFront Client
+
+async function cloudFrontInvalidate(paths) {
+  const distributionId = process.env.DISTRIBUTION_ID; // Ensure DISTRIBUTION_ID is set in your environment
```
Comment (aps08) [inline server/utils/cloudFront.js]: use multiline comments at top of the function,

**Review by aps08 — CHANGES_REQUESTED** [review]
Add it in the reupload option for image for admin.

**Comment by amoghar29** [comment]
Have made all the said changes.

File: server/controllers/admin/reupload.js [hunk 2]
```diff
@@ -91,6 +92,12 @@ async function adminReUploadController(req, res, next) {
         message: "Failed to update record",
       });
     }
+    try {
```
Comment (aps08) [inline server/controllers/admin/reupload.js]: why try catch is here ? It is already at the start.

Comment (amoghar29) [reply server/controllers/admin/reupload.js]: removed.

**Review by aps08 — APPROVED** [review]

**Comment by aps08** [comment]
@amoghar29 get is approved by one more person.

**Review by DeltaDynamo — APPROVED** [review]

**Pull request merged**